src/library_wasmfs_js_file.js

```diff
@@ -0,0 +1,53 @@
+mergeInto(LibraryManager.library, {
+  // JSFile backend: Store a file's data in JS. We map File objects in C++ to
+  // entries here that contain typed arrays.
+  $wasmFS$JSMemoryFiles: {},
+
+  _wasmfs_backend_add_js_file__deps: [
+    '$wasmFS$backends',
+    '$wasmFS$JSMemoryFiles',
+  ],
+  _wasmfs_backend_add_js_file: function(backend) {
```

Comment:
Should this be `add_js_backend` rather than `add_js_file`?

Reply:
I was trying to be consistent, `wasmfs_backend_*` are APIs to add various backends. This backend is called the "js file backend".

Reply:
Hmm, maybe we should change the name of the backend, or maybe change the naming convention. `_wasmfs_add_js_file_backend` would be much clearer that we are adding a backend rather than a `File`.

Reply:
How about `wasmfs_backend_add_jsfile`? The lack of a `_` makes it clear that "jsfile" is a "thing".

Reply:
Yeah, that seems ok as well.

Reply:
Actually, this is the internal API, so we might as well change it as much as we want I guess. Also we have a public API `wasmfs_create_js_file_backend` so I'll make it more similar to that, and follow the convention of duplicating the public C API + adding `_js` to the end.